libs/infinity_emb/tests/end_to_end/test_torch_reranker.py

```diff
@@ -89,6 +90,61 @@ async def test_reranker(client, model_base, helpers):
     for i, pred in enumerate(predictions):
         assert abs(rdata_results[i]["relevance_score"] - pred["score"]) < 0.01
 
+@pytest.mark.anyio
+async def test_reranker_top_k(client):
+    query = "Where is the Eiffel Tower located?"
+    documents = [
```

Comment:
Potential flaw: If backend does not do sorting, top_k=1 will just take always the first result.

Better unit test: Wrap this with a unit test, and `return_text`. Make sure that topk=1 solution is always paris in this unit test. e.g. use https://docs.python.org/3/library/itertools.html
````
# someting like:
for return_text in [true, False]
    for raw_score in [True, False]:
          for permutation in itertools.permutation([..paris, ..us, uk]):
               # you test above
```